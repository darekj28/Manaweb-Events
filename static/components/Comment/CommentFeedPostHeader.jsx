var React = require('react');
export default class CommentFeedPostHeader extends React.Component {
	render() {
		return(
			<div id = {"comment_" + this.props.comment.unique_id} className="FeedPostHeader">
				{this.props.comment.name != undefined && 
				<div className="headerpart pull-left"><b>{this.props.comment.name}</b></div> }
				{this.props.comment.name != undefined && 
				<div className="headerpart pull-left username text-muted">@{this.props.comment.userID}</div>}
				{!this.props.isOriginalPost && 
				<div className="headerpart time pull-left text-muted">&#8226; {this.props.comment.time}</div>}
			</div>
		)
	}
}