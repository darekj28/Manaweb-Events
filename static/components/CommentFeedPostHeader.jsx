var React = require('react');
export default class CommentFeedPostHeader extends React.Component {
	constructor(props) {
		super(props);
	}
	render() {
		return(
			<div className="FeedPostHeader">
				<div className="pull-left name"><b>{this.props.name}</b></div>
				<div className="pull-left username text-muted">@{this.props.userID}</div>
				<div className="time pull-left text-muted">&#8226; {this.props.time}</div>
				<div className="pull-right">
					{!this.props.isOriginalPost && <div className="dropdown">
						<a href="#" className="dropdown-toggle" data-toggle="dropdown">
			                <span className="glyphicon glyphicon-option-horizontal 
			                				pull-left CommentBottomIcon AppGlyphicon"></span>
			            </a>
			            <ul className="CommentDropdown pull-right dropdown-menu">
			              	{(this.props.isOP || this.props.isAdmin) && <li><a href="#" onClick={this.props.handleCommentEdit}>Edit comment</a></li> }
			              	{(this.props.isOP || this.props.isAdmin) && <li><a href="#" onClick={this.props.handleCommentDelete}>Delete comment</a></li> }
			              	{!(this.props.isOP || this.props.isAdmin) && <li><a href="#" onClick={this.props.handleCommentReport}>Report comment</a></li> }
			            </ul>
		        	</div>}
				</div>
			</div>
			)
	}
}