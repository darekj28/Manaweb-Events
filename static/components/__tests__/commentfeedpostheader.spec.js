import CommentFeedPostHeader from "../CommentFeedPostHeader.jsx";

describe("CommentFeedPostHeader", () => {
	it("renders correctly for isOP", () => {
		const comment = {
			name : "a",
			userID : "a",
			time : "a"
		};
		const cfph = renderer.create(
			<CommentFeedPostHeader name={comment.name} userID={comment.userID} time={comment.time}
				isOP={true} isAdmin={false} handleCommentEdit={jest.fn()}
				handleCommentDelete={jest.fn()} handleCommentReport={jest.fn()}
				isOriginalPost={false}/>
			);
		expect(cfph).toMatchSnapshot();
	});
	it("renders correctly for isAdmin", () => {
		const comment = {
			name : "a",
			userID : "a",
			time : "a"
		};
		const cfph = renderer.create(
			<CommentFeedPostHeader name={comment.name} userID={comment.userID} time={comment.time}
				isOP={false} isAdmin={true} handleCommentEdit={jest.fn()}
				handleCommentDelete={jest.fn()} handleCommentReport={jest.fn()}
				isOriginalPost={false}/>
			);
		expect(cfph).toMatchSnapshot();
	});
	it("renders correctly for not isOP or isAdmin", () => {
		const comment = {
			name : "a",
			userID : "a",
			time : "a"
		};
		const cfph = renderer.create(
			<CommentFeedPostHeader name={comment.name} userID={comment.userID} time={comment.time}
				isOP={false} isAdmin={false} handleCommentEdit={jest.fn()}
				handleCommentDelete={jest.fn()} handleCommentReport={jest.fn()}
				isOriginalPost={false}/>
			);
		expect(cfph).toMatchSnapshot();
	});
	it("renders correctly for original post", () => {
		const comment = {
			name : "a",
			userID : "a",
			time : "a"
		};
		const cfph = renderer.create(
			<CommentFeedPostHeader name={comment.name} userID={comment.userID} time={comment.time}
				isOP={false} isAdmin={false} handleCommentEdit={jest.fn()}
				handleCommentDelete={jest.fn()} handleCommentReport={jest.fn()}
				isOriginalPost={true}/>
			);
		expect(cfph).toMatchSnapshot();
	});
});