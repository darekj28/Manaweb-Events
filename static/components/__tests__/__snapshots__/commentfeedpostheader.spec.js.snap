exports[`CommentFeedPostHeader renders correctly for isAdmin 1`] = `
<div
  className="FeedPostHeader">
  <div
    className="pull-left name">
    <b>
      a
    </b>
  </div>
  <div
    className="pull-left username text-muted">
    @
    a
  </div>
  <div
    className="time pull-left text-muted">
    • 
    a
  </div>
  <div
    className="pull-right">
    <div
      className="dropdown">
      <a
        className="dropdown-toggle"
        data-toggle="dropdown"
        href="#">
        <span
          className="glyphicon glyphicon-option-horizontal  pull-left CommentBottomIcon AppGlyphicon" />
      </a>
      <ul
        className="CommentDropdown pull-right dropdown-menu">
        <li>
          <a
            href="#"
            onClick={[Function]}>
            Edit comment
          </a>
        </li>
        <li>
          <a
            href="#"
            onClick={[Function]}>
            Delete comment
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`CommentFeedPostHeader renders correctly for isOP 1`] = `
<div
  className="FeedPostHeader">
  <div
    className="pull-left name">
    <b>
      a
    </b>
  </div>
  <div
    className="pull-left username text-muted">
    @
    a
  </div>
  <div
    className="time pull-left text-muted">
    • 
    a
  </div>
  <div
    className="pull-right">
    <div
      className="dropdown">
      <a
        className="dropdown-toggle"
        data-toggle="dropdown"
        href="#">
        <span
          className="glyphicon glyphicon-option-horizontal  pull-left CommentBottomIcon AppGlyphicon" />
      </a>
      <ul
        className="CommentDropdown pull-right dropdown-menu">
        <li>
          <a
            href="#"
            onClick={[Function]}>
            Edit comment
          </a>
        </li>
        <li>
          <a
            href="#"
            onClick={[Function]}>
            Delete comment
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`CommentFeedPostHeader renders correctly for not isOP or isAdmin 1`] = `
<div
  className="FeedPostHeader">
  <div
    className="pull-left name">
    <b>
      a
    </b>
  </div>
  <div
    className="pull-left username text-muted">
    @
    a
  </div>
  <div
    className="time pull-left text-muted">
    • 
    a
  </div>
  <div
    className="pull-right">
    <div
      className="dropdown">
      <a
        className="dropdown-toggle"
        data-toggle="dropdown"
        href="#">
        <span
          className="glyphicon glyphicon-option-horizontal  pull-left CommentBottomIcon AppGlyphicon" />
      </a>
      <ul
        className="CommentDropdown pull-right dropdown-menu">
        <li>
          <a
            href="#"
            onClick={[Function]}>
            Report comment
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`CommentFeedPostHeader renders correctly for original post 1`] = `
<div
  className="FeedPostHeader">
  <div
    className="pull-left name">
    <b>
      a
    </b>
  </div>
  <div
    className="pull-left username text-muted">
    @
    a
  </div>
  <div
    className="time pull-left text-muted">
    • 
    a
  </div>
  <div
    className="pull-right" />
</div>
`;
